statsd/udp.go

```diff
@@ -7,26 +7,27 @@ import (
 
 // udpWriter is an internal class wrapping around management of UDP connection
 type udpWriter struct {
-	conn net.Conn
+	conn net.PacketConn
+	addr string
 }
 
 // New returns a pointer to a new udpWriter given an addr in the format "hostname:port".
 func newUDPWriter(addr string, _ time.Duration) (*udpWriter, error) {
-	udpAddr, err := net.ResolveUDPAddr("udp", addr)
+	conn, err := net.ListenPacket("udp", ":0")
 	if err != nil {
 		return nil, err
 	}
-	conn, err := net.DialUDP("udp", nil, udpAddr)
-	if err != nil {
-		return nil, err
-	}
-	writer := &udpWriter{conn: conn}
+	writer := &udpWriter{conn: conn, addr: addr}
 	return writer, nil
 }
 
 // Write data to the UDP connection with no error handling
 func (w *udpWriter) Write(data []byte) (int, error) {
-	return w.conn.Write(data)
+	dst, err := net.ResolveUDPAddr("udp", w.addr)
```

Comment:
This is an expensive operation that you do not want to do before every packet write.  If you want to do such a change, it needs to be done infrequently to avoid substantially increasing the overhead of the datadog statsd client.